crates/core/src/kernel/snapshot/log_segment.rs

```diff
@@ -30,9 +30,12 @@ lazy_static! {
     pub(super) static ref COMMIT_SCHEMA: StructType = StructType::new(vec![
         ActionType::Add.schema_field().clone(),
         ActionType::Remove.schema_field().clone(),
+        ActionType::Txn.schema_field().clone(),
+    ]);
+    pub(super) static ref CHECKPOINT_SCHEMA: StructType = StructType::new(vec![
+        ActionType::Add.schema_field().clone(),
+        ActionType::Txn.schema_field().clone(),
```

Comment:
would it maybe make sense to make this configurable? The goal is to be as minimal as possible when reading the log and eventuially update the parquet reader to leverage push-down as best we can.

We will always require add / remove but can maybe read txns and others based on config. Another alternative would maybe be to just read the transactions separately and in another PR add a simple caching layer for reading especially the json files?

What do you think?